File: Python/1040.py
x = input().split()
n1, n2, n3, n4 = x
m = (float(n1) * 2 + float(n2) * 3 + float(n3) * 4 + float(n4) * 1) / 10
print('Media: {:.1f}'.format(m))
if m >= 7.0:
    print('Aluno aprovado.')
if m < 5.0:
    print('Aluno reprovado.')
if 5.0 <= m <= 6.9:
    print('Aluno em exame.')
    y = float(input())
    print('Nota do exame: {}'.format(y))
    m2 = (y + m) / 2
    if m2 >=5:
        print('Aluno aprovado.')
        print('Media final: {:.1f}'.format(m2))
    else:
        print('Aluno reprovado.')
        print('Media final: {:.1f}'.format(m2))
    

File: C#/1040.cs
using System;
using System.Linq;
using System.Globalization;

class Program {
    static void Main() 
    {
        CultureInfo.DefaultThreadCurrentCulture = new CultureInfo("en-US");
        decimal[] x = Console.ReadLine().Split().Select(decimal.Parse).ToArray();
        decimal media = Math.Floor((x[0] * 2 + x[1] * 3 + x[2] * 4 + x[3] * 1) / 10m * 10) / 10; 
        Console.WriteLine($"Media: {media:F1}");
        if (media >= 7)
        {
            Console.WriteLine("Aluno aprovado.");
        }
        else if (media < 5)
        {
            Console.WriteLine("Aluno reprovado.");
        }
        else
        {
            Console.WriteLine("Aluno em exame.");
            decimal notaExame = decimal.Parse(Console.ReadLine());
            Console.WriteLine($"Nota do exame: {notaExame:F1}");
            decimal mediaFinal = (notaExame + media) / 2;
            if (mediaFinal >= 5)
            {
                Console.WriteLine("Aluno aprovado.");
            }
            else
            {
                Console.WriteLine("Aluno reprovado.");
            }
            Console.WriteLine($"Media final: {mediaFinal:F1}");
            
        }
    }
}
